spec/factory_logger_spec.rb

```diff
@@ -94,4 +94,30 @@
                                             ]
     end
   end
+
+  describe '.descendants' do
+    let!(:top) { create :factory_log, name: "top", parent_id: nil }
+    let!(:child1) { create :factory_log, name: "child1", parent_id: top.id }
+    let!(:child2) { create :factory_log, name: "child2", parent_id: top.id }
+    let!(:child3) { create :factory_log, name: "child3", parent_id: top.id }
+    let!(:grand_son1_1) { create :factory_log, name: "grand_son1_1", parent_id: child1.id }
+    let!(:grand_son1_2) { create :factory_log, name: "grand_son1_2", parent_id: child1.id }
+    let!(:grand_son2_1) { create :factory_log, name: "grand_son2_1", parent_id: child2.id }
+
+    it 'have tree' do
```

Comment:
Style/StringLiterals: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.